### PR TITLE
Require that single-action examples are not duplicated and specify exactly one service and one action

### PR DESCRIPTION
**Pull request by Laren-AWS**
This is the first part of validation needed for canonical title update:

- Validate that API examples have exactly one service and one action.
- Check for duplicate API examples by svc:action combo.
- Update tests to pass with canonical name validation updates.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws_doc_sdk_examples_tools/metadata.py [hunk 1]
```diff
@@ -393,6 +406,24 @@ def parse(
     return examples, errors
 
 
+def validate_no_duplicate_api_examples(examples: List[Example], errors: MetadataErrors):
```
Comment (DavidSouther) [inline aws_doc_sdk_examples_tools/metadata.py:409]: This doesn't need to be List, just Iterable. (then no need to build a list when calling just to rip it apart again below)

File: aws_doc_sdk_examples_tools/metadata.py [hunk 2]
```diff
@@ -393,6 +406,24 @@ def parse(
...
+    """Call this on a full set of examples to verify that there are no duplicate API examples."""
+    svc_action_map = defaultdict(list)
```
Comment (DavidSouther) [inline aws_doc_sdk_examples_tools/metadata.py:411]: Should this be `defaultdict(set)`?

**Review by DavidSouther — CHANGES_REQUESTED** [review]

File: aws_doc_sdk_examples_tools/metadata.py (hunk 1 above)
Comment (Laren-AWS) [reply aws_doc_sdk_examples_tools/metadata.py:409]: Interfaces over classes, yes! Python makes me lazy...

File: aws_doc_sdk_examples_tools/metadata.py (hunk 2 above)
Comment (Laren-AWS) [reply aws_doc_sdk_examples_tools/metadata.py]: Makes more sense, yes.

**Review by DavidSouther — APPROVED** [review]

**Pull request merged**